src/Core/Generator/SchemaGenerator.cs

```diff
@@ -0,0 +1,336 @@
+        /// <returns>A string representing the generated GraphQL schema.</returns>
+        /// <exception cref="InvalidOperationException">Thrown if the JSON data is empty or the container name is null or empty.</exception>
+        public static string Generate(List<JsonDocument> jsonData, string containerName, RuntimeConfig? config = null)
+        {
+            // Validate input parameters.
+            if (jsonData == null || jsonData.Count == 0 || string.IsNullOrEmpty(containerName))
+            {
+                throw new InvalidOperationException("JArray must contain at least one JSON object and Container Name can not be blank");
+            }
+
+            // Create an instance of SchemaGenerator and generate the schema.
+            return new SchemaGenerator(jsonData, containerName, config)
+                        .ConvertJsonToGQLSchema();
+        }
+
+        /// <summary>
+        /// Converts the JSON data into a GraphQL schema string.
+        /// </summary>
+        /// <returns>A string representing the GraphQL schema.</returns>
+        /// <exception cref="InvalidOperationException">Thrown if the JSON data contains non-object elements.</exception>
+        private string ConvertJsonToGQLSchema()
+        {
+            // Process each JSON document to build the GraphQL schema.
+            foreach (JsonDocument token in _data)
+            {
+                if (token is JsonDocument jsonToken)
+                {
+                    TraverseJsonObject(jsonToken, _containerName);
+                }
+                else
+                {
+                    throw new InvalidOperationException("Invalid JsonDocument found.");
+                }
+            }
+
+            // Generate the GraphQL schema string from the collected entity information.
+            return GenerateGQLSchema();
+        }
+
+        /// <summary>
+        /// Generates a GraphQL schema string from the collected attributes and entities.
+        /// </summary>
+        /// <returns>A string representing the GraphQL schema.</returns>
+        private string GenerateGQLSchema()
+        {
+            StringBuilder sb = new();
+
+            // Iterate through the collected entities and their attributes to build the schema.
+            foreach (KeyValuePair<string, HashSet<AttributeObject>> entity in _attrMapping)
+            {
+                // Determine if the entity is the root entity.
+                bool isRoot = entity.Key == _containerName.Pascalize();
+
+                sb.Append($"type {entity.Key} ");
+
+                // Append model directive if applicable.
+                if (_entityAndSingularNameMapping.ContainsKey(entity.Key) && _entityAndSingularNameMapping[entity.Key] != entity.Key)
+                {
+                    sb.Append($"@model(name:\"{_entityAndSingularNameMapping[entity.Key]}\") ");
+                }
+                else if (isRoot)
+                {
+                    sb.Append("@model ");
```

Comment:
I though we no longer allow `@model(name=)` without a name value. Is that not accurate?